src/routes/InstancesRoute.test.js

```diff
@@ -0,0 +1,315 @@
+  hasPerm: () => true,
+  logger: { log: noop },
+  locale: 'en-US',
+  plugins: {},
+};
+
+const InstancesRouteSetup = ({
+  instances = instancesFixture,
+  sendCallout = noop,
+  quickExportPOST = noop,
+} = {}) => (
+  <Router>
+    <StripesContext.Provider value={stripesStub}>
+      <CalloutContext.Provider value={{ sendCallout }}>
+        <DataContext.Provider value={{
+          contributorTypes: [],
+          instanceTypes: [],
+          locations: [],
+          instanceFormats: [],
+          modesOfIssuance: [],
+          natureOfContentTerms: [],
+          tagsRecords: [],
+        }}
+        >
+          <Paneset>
+            <Layer
+              isOpen
+              contentLabel="label"
+            >
+              <OverlayContainer />
+              <InstancesRoute
+                resources={{
+                  query: {
+                    query: '',
+                    sort: 'title',
+                  },
+                  records: {
+                    hasLoaded: true,
+                    resource: 'records',
+                    records: instances,
+                    other: { totalRecords: instances.length },
+                  },
+                  resultCount: instances.length,
+                  resultOffset: 0,
+                }}
+                mutator={{
+                  quickExport: { POST: quickExportPOST },
+                  resultCount: { replace: noop },
+                }}
+              />
+            </Layer>
+          </Paneset>
+        </DataContext.Provider>
+      </CalloutContext.Provider>
+    </StripesContext.Provider>
+  </Router>
+);
+
+describe('InstancesRoute', () => {
+  const quickExportAPICallMock = jest.fn(Promise.resolve.bind(Promise));
+  const sendCalloutMock = jest.fn();
+  let renderWithIntlResult;
+
+  describe('rendering InstancesRoute', () => {
+    beforeEach(() => {
+      renderWithIntlResult = renderWithIntl(
+        <InstancesRouteSetup
+          quickExportPOST={quickExportAPICallMock}
+          sendCallout={sendCalloutMock}
+        />,
+        translationsProperties
+      );
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should have proper list results size', () => {
+      expect(document.querySelectorAll('#pane-results-content .mclRowContainer > [role=row]').length).toEqual(3);
+    });
+
+    it('should render nothing for select row column header', () => {
+      expect(document.querySelector('#pane-results-content .mclHeaderRow > [role=columnheader]:first-child').textContent).toEqual('');
+    });
+
+    it('should display unchecked select row checkbox', () => {
+      const selectRowCheckboxes = screen.getAllByRole('checkbox', { name: 'Select instance' });
+
+      expect(selectRowCheckboxes[0]).not.toBeChecked();
+    });
+
+    it('should not display warning about exceeded quick export limit in the action menu', () => {
+      userEvent.click(screen.getByRole('button', { name: 'Actions' }));
```

Comment:
Maybe it'll be better to put all tests, where action menu is being opened in separate `describe` block with `befereEach` button click?

Reply:
Makes sense.